File: complete-backend/src/main/java/com/yuwan/completebackend/model/entity/School.java
package com.yuwan.completebackend.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.util.Date;

/**
 * 学校信息实体类
 * 对应数据库表 school_info
 *
 * @author 系统架构师
 * @version 1.0
 * @since 2026-02-22
 */
@Data
@TableName("school_info")
@Schema(description = "学校信息实体")
public class School implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    @TableId(value = "school_id", type = IdType.ASSIGN_ID)
    @Schema(description = "学校ID")
    private String schoolId;

    @Schema(description = "学校名称")
    private String schoolName;

    @Schema(description = "学校编码")
    private String schoolCode;

    @Schema(description = "详细地址")
    private String address;

    @Schema(description = "联系人")
    private String contactPerson;

    @Schema(description = "联系电话")
    private String contactPhone;

    @Schema(description = "学校邮箱")
    private String schoolEmail;

    @Schema(description = "学校简介")
    private String description;

    @Schema(description = "状态（0-禁用 1-正常）")
    private Integer schoolStatus;

    @TableLogic
    @Schema(description = "逻辑删除（0-未删除 1-已删除）")
    private Integer deleted;

    @TableField(fill = FieldFill.INSERT)
    @Schema(description = "创建时间")
    private Date createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    @Schema(description = "更新时间")
    private Date updateTime;
}

File: complete-frontend/src/types/school.ts
/**
 * 学校管理相关类型定义
 * @description 包含学校信息、查询参数、DTO等类型
 * @author YuWan
 * @date 2026-02-22
 */

/**
 * 学校状态枚举
 */
export enum SchoolStatus {
  /** 禁用 */
  DISABLED = 0,
  /** 正常 */
  ACTIVE = 1
}

/**
 * 学校状态标签映射
 */
export const SCHOOL_STATUS_LABELS: Record<number, string> = {
  [SchoolStatus.DISABLED]: '禁用',
  [SchoolStatus.ACTIVE]: '正常'
}

/**
 * 学校状态颜色映射
 */
export const SCHOOL_STATUS_COLORS: Record<number, string> = {
  [SchoolStatus.DISABLED]: 'red',
  [SchoolStatus.ACTIVE]: 'green'
}

/**
 * 学校下拉选项 VO（精简版，所有登录用户可用）
 */
export interface SchoolOptionVO {
  /** 学校ID */
  schoolId: string
  /** 学校名称 */
  schoolName: string
}

/**
 * 学校信息 VO
 */
export interface SchoolVO {
  /** 学校ID */
  schoolId: string
  /** 学校名称 */
  schoolName: string
  /** 学校编码 */
  schoolCode?: string
  /** 详细地址 */
  address?: string
  /** 联系人 */
  contactPerson?: string
  /** 联系电话 */
  contactPhone?: string
  /** 学校邮箱 */
  schoolEmail?: string
  /** 学校简介 */
  description?: string
  /** 状态（0-禁用 1-正常） */
  schoolStatus: number
  /** 状态描述 */
  schoolStatusDesc?: string
  /** 创建时间 */
  createTime: string
  /** 更新时间 */
  updateTime?: string
}

/**
 * 学校查询参数 QueryVO
 */
export interface SchoolQueryVO {
  /** 学校名称（模糊查询） */
  schoolName?: string
  /** 学校编码 */
  schoolCode?: string
  /** 状态（0-禁用 1-正常） */
  schoolStatus?: number
  /** 页码 */
  pageNum?: number
  /** 每页大小 */
  pageSize?: number
}

/**
 * 创建学校请求 DTO
 */
export interface CreateSchoolDTO {
  /** 学校名称 */
  schoolName: string
  /** 学校编码 */
  schoolCode?: string
  /** 详细地址 */
  address?: string
  /** 联系人 */
  contactPerson?: string
  /** 联系电话 */
  contactPhone?: string
  /** 学校邮箱 */
  schoolEmail?: string
  /** 学校简介 */
  description?: string
}

/**
 * 更新学校请求 DTO
 */
export interface UpdateSchoolDTO {
  /** 学校名称 */
  schoolName?: string
  /** 学校编码 */
  schoolCode?: string
  /** 详细地址 */
  address?: string
  /** 联系人 */
  contactPerson?: string
  /** 联系电话 */
  contactPhone?: string
  /** 学校邮箱 */
  schoolEmail?: string
  /** 学校简介 */
  description?: string
  /** 状态（0-禁用 1-正常） */
  schoolStatus?: number
}
